machine_learning_hep/multiprocesser.py

```diff
@@ -136,28 +157,36 @@ def __init__(self, case, proc_class, datap, typean, run_param, mcordata):
                 self.lper_fileresp.append(os.path.join(direc, self.n_fileresp))
 
     def multi_unpack_allperiods(self):
-        for indexp in range(self.prodnumber):
+        for indexp, _ in enumerate(self.process_listsample):
```

Comment:
Why do you use enumerate if you then don't use it?
```suggestion
        for indexp in range(len(self.process_listsample)):
```